model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala

```diff
@@ -35,10 +46,141 @@ object FocalPlane {
     Enumerated.of(SingleSlit, MultipleSlit, IFU)
 }
 
+trait InstrumentRow {
+  def instrument: Instrument
+
+  type Disperser
+  val disperser: Disperser
```

Comment:
I'm not familiar with doing this kind of thing.  What does having abstract types like this do for us over having instrument-specific `InstrumentRow` types in an ADT?   Instead of matching on the `InstrumentRow` you match on the different `Disperser` types but in that case there are no guarantees about what type possibilities may exist.

Nothing prevents you from having a GMOS `Filter` and an F2 `Disperser` for example?

If I have an `InstrumentRow` containing a GMOS longslit definition, how do I turn it into a GMOS configuration?

Reply:
I was hoping you had an answer for how to turn this into a config. 😀